cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java

```diff
@@ -0,0 +1,170 @@
+
+        this.deserializationFacade = deserializationFacade;
+        this.serializationFacade = serializationFacade;
+        this.credentialsProvider = credentialsProvider;
+        this.deserializer = deserializerImpl;
+        this.serializer = serializerImpl;
+    }
+
+    /**
+     * Configure the MM2 Schema Replication Converter.
+     * @param configs configuration elements for the converter
+     * @param isKey true if key, false otherwise
+     */
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+        this.isKey = isKey;
+
+        credentialsProvider = DefaultCredentialsProvider.builder().build();
+
+        // Put the source and target regions into configurations respectively
+        Map<String, Object> sourceConfigs = new HashMap<>(configs);
+        Map<String, Object> targetConfigs = new HashMap<>(configs);
+
+
+        if (configs.get(AWSSchemaRegistryConstants.AWS_SOURCE_REGION) == null){
+            throw new DataException("Source Region is not provided.");
+        }
+        else if (configs.get(AWSSchemaRegistryConstants.AWS_TARGET_REGION) == null && configs.get(AWSSchemaRegistryConstants.AWS_REGION) == null){
+            throw new DataException("Target Region is not provided.");
+        }
+
+        sourceConfigs.put(AWSSchemaRegistryConstants.AWS_REGION, configs.get(AWSSchemaRegistryConstants.AWS_SOURCE_REGION));
+        targetConfigs.put(AWSSchemaRegistryConstants.AWS_REGION, configs.get(AWSSchemaRegistryConstants.AWS_TARGET_REGION));
+
+        deserializationFacade =
+                GlueSchemaRegistryDeserializationFacade.builder()
+                        .credentialProvider(credentialsProvider)
+                        .configs(sourceConfigs)
+                        .build();
+
+        serializationFacade   =
+                GlueSchemaRegistrySerializationFacade.builder()
+                        .credentialProvider(credentialsProvider)
+                        .configs(targetConfigs)
+                        .build();
+
+        serializer   = new GlueSchemaRegistrySerializerImpl(credentialsProvider, new GlueSchemaRegistryConfiguration(targetConfigs));
+
+        deserializer = new GlueSchemaRegistryDeserializerImpl(credentialsProvider, new GlueSchemaRegistryConfiguration(sourceConfigs));
+
+    }
+
+    @Override
+    public byte[] fromConnectData(String topic, org.apache.kafka.connect.data.Schema schema, Object value) {
+        byte[] bytes = (byte[]) value;
+        if (value == null) return new byte[0];
+
+        try {
+            byte[] deserializedBytes = deserializer.getData(bytes);
+            Schema deserializedSchema = deserializer.getSchema(bytes);
+
+            byte[] encodedByte = serializer.encode(null, deserializedSchema, deserializedBytes);
+            com.amazonaws.services.schemaregistry.common.Schema returnedSchema = getSchema(bytes);
+            UUID uuid = registerSchema(returnedSchema);
```

Comment:
Isn't this automatically done by the `serializer`? [Ref](https://github.com/awslabs/aws-glue-schema-registry/blob/master/serializer-deserializer/src/main/java/com/amazonaws/services/schemaregistry/serializers/GlueSchemaRegistrySerializationFacade.java#L112) Why do we have to do it again?